contracts/src/v0.8/dev/ArbitrumSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,286 @@
+    }
+  }
+
+  /**
+   * @dev Returns an AggregatorV2V3Interface compatible answer from status flag
+   *
+   * @param status The status flag to convert to an aggregator-compatible answer
+   */
+  function getStatusAnswer(bool status) private pure returns (int256) {
+    return status ? int256(1) : int256(0);
+  }
+
+  /**
+   * @notice The L2 xDomain `msg.sender`, generated from L1 sender address
+   */
+  function crossDomainMessenger() public view returns (address) {
+    return AddressAliasHelper.applyL1ToL2Alias(l1Sender());
+  }
+
+  /**
+   * @notice Raise or lower the flag on the stored Flags contract.
+   */
+  function forwardStatusToFlags(bool status) private {
+    if (status) {
+      FLAGS.raiseFlag(FLAG_L2_SEQ_OFFLINE);
+    } else {
+      FLAGS.lowerFlag(FLAG_L2_SEQ_OFFLINE);
+    }
+  }
+
+  /**
+   * @notice Helper function to record a round and set the latest feed state.
+   *
+   * @param roundId The round ID to record
+   * @param status Sequencer status
+   * @param timestamp Block timestamp of status update
+   */
+  function recordRound(
+    uint80 roundId,
+    bool status,
+    uint64 timestamp
+  ) private {
+    Round memory nextRound = Round(status, timestamp);
+    FeedState memory feedState = FeedState(roundId, status, timestamp);
+
+    s_rounds[roundId] = nextRound;
+    s_feedState = feedState;
+
+    emit NewRound(roundId, msg.sender, timestamp);
+    emit AnswerUpdated(getStatusAnswer(status), roundId, timestamp);
+  }
+
+  /**
+   * @notice Record a new status and timestamp if it has changed since the last round.
+   * @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+   *
+   * @param status Sequencer status
+   * @param timestamp Block timestamp of status update
+   */
+  function updateStatus(bool status, uint64 timestamp) external override {
+    FeedState memory feedState = s_feedState;
+    require(feedState.latestRoundId > 0, "ArbitrumSequencerUptimeFeed has not been initialized");
+    require(msg.sender == crossDomainMessenger(), "Sender is not the L2 messenger");
+
+    // Ignore if status did not change
+    if (feedState.latestStatus == status) {
+      return;
+    }
+
+    // Prepare a new round with updated status
+    feedState.latestRoundId += 1;
+    recordRound(feedState.latestRoundId, status, timestamp);
+
+    forwardStatusToFlags(status);
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestAnswer() external view override checkAccess returns (int256) {
+    FeedState memory feedState = s_feedState;
+    require(feedState.latestRoundId > 0, "ArbitrumSequencerUptimeFeed has not been initialized");
+    return getStatusAnswer(feedState.latestStatus);
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestTimestamp() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    require(feedState.latestRoundId > 0, "ArbitrumSequencerUptimeFeed has not been initialized");
```

Comment:
```suggestion
    require(feedState.latestRoundId > 0, "ArbitrumSequencerUptimeFeed has not been initialized");
```
This line is duplicated in a lot of places. Let's pull it out into a helper method so that we can more easily modify it in the future.

Reply:
Factored out to a modifier `ensureInitialized`.